src/tes/server/task_boltdb.go

```diff
@@ -343,11 +344,10 @@ func (taskBolt *TaskBolt) GetServiceInfo(ctx context.Context, info *ga4gh_task_e
 	//     For example, you can't have multiple S3 endpoints
 	out := map[string]string{}
 	for _, i := range taskBolt.serverConfig.Storage {
-		if i.Local != nil {
+		if reflect.DeepEqual(i.Local, tes.LocalStorage{}) {
 			out["Local.AllowedDirs"] = strings.Join(i.Local.AllowedDirs, ",")
 		}
-
-		if i.S3 != nil {
+		if reflect.DeepEqual(i.S3, tes.S3Storage{}) {
```

Comment:
This could do `i.S3.Endpoint != ""`. 

Optionally it could also check the Key and Secret fields, which would provide some extra config validation. I don't know if key and secret are ever optional in S3 though.